docs/source/module_configuration.rst

```diff
@@ -158,7 +158,7 @@ and/or for a group of modules:
       - modules:
           - path: mytfmodule
           - path: anothermytfmodule
-        module_options:  # shared between all modules in deployment
+        module-options:  # shared between all modules in deployment
           terraform_version:
```

Comment:
Should these options support `-/_` as well? It _could_ get confusing knowing which to use if they both work in some places but not all.

Reply:
@ITProKyle As it stands from this PR, there will be some consistency in:
* All top-level & deployment options are documented with hyphens, but support being used with either hyphens or underscores.
* All module options only support underscores (and haven't been documented or used with hyphens in the past)

I'm hesitant to make a practice/requirement of having the various module options support dual-format options as well, but don't love the current situation either.

Maybe it's better to rip off the bandaid and change the top level ones (`account-id`->`account_id`, etc), autocorrecting hyphens to underscores for now (with a deprecation warning about future removal). Thoughts?

Reply:
We could also bias everything toward underscores instead of hyphens so everything is consistent in the docs and leave top level the only thing that can also handle hyphens. We can leave this out of the docs as a "hidden feature" since its YMMV. This may suffice to mitigate confusion of what can/can't do it without having to drop support for things that use hyphens right now.

Essentially your idea of "rip off the bandaid" but not deprecating support for hyphens at the top level. At least not at this point. It can always be revisited in a few months.